src/media/js/views/search.js

```diff
@@ -215,6 +215,13 @@ define('views/search',
             pageTypes += ' leaf';
         }
 
+        if (query && queryParam === 'q') {
```

Comment:
We can probably change this block to be `if (query) {` and inside do `switch queryParam {case 'author': ... case 'q':` etc.